pkgs/development/libraries/mesa/default.nix

```diff
@@ -119,7 +120,8 @@ self = stdenv.mkDerivation {
     "-Dgallium-opencl=icd" # Enable the gallium OpenCL frontend
     "-Dclang-libdir=${llvmPackages.clang-unwrapped.lib}/lib"
   ] ++ optional enablePatentEncumberedCodecs
-    "-Dvideo-codecs=h264dec,h264enc,h265dec,h265enc,vc1dec";
+    "-Dvideo-codecs=h264dec,h264enc,h265dec,h265enc,vc1dec"
+  ++ lib.optionals (vulkanLayers != []) [ "-D vulkan-layers=${builtins.concatStringsSep "," vulkanLayers}" ];
```

Comment:
You could simplify it to use `optional`.

Reply:
Done